submission/file-sources/src/main/kotlin/uk/ac/ebi/io/sources/FtpSource.kt

```diff
@@ -42,14 +45,24 @@ class FtpSource(
         }
     }
 
-    override suspend fun getFileList(path: String): File? {
-        return findFile(path)?.let { downloadFile(ftpUrl.resolve(path)) }
-    }
+    override suspend fun getFileList(path: String): File? = findFile(path)?.let { downloadFile(ftpUrl.resolve(path)) }
 
+    @Suppress("TooGenericExceptionCaught")
     private fun findFile(filePath: String): FTPFile? {
-        val ftpPath = ftpUrl.resolve(filePath)
-        val files = ftpClient.listFiles(ftpPath.parent)
-        return files.firstOrNull { it.name == ftpPath.name }
+        var attempt = 0
+
+        fun ftpResolve(): FTPFile? {
```

Comment:
I think what we should do here which I am aware increase the scope of ticket.  is to wrap ftpClient calls using SuspendRetryTemplate just as we did with `RetryWebClient` also I just notice FTP operaiton are not flag with suspend and  withContext(Dispatchers.IO)  (as they are blocking so we kind of need ot fix this too)